src/PowerShellEditorServices/Services/DebugAdapter/BreakpointService.cs

```diff
@@ -0,0 +1,412 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Management.Automation.Language;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.PowerShell.EditorServices.Logging;
+using Microsoft.PowerShell.EditorServices.Services.DebugAdapter;
+using Microsoft.PowerShell.EditorServices.Utility;
+
+namespace Microsoft.PowerShell.EditorServices.Services
+{
+    internal class BreakpointService
+    {
+        private const string s_psesGlobalVariableNamePrefix = "__psEditorServices_";
+        private readonly ILogger<BreakpointService> _logger;
+        private readonly PowerShellContextService _powerShellContextService;
+
+        private readonly ConcurrentDictionary<string, List<Breakpoint>> _breakpointsPerFile =
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private field '_breakpointsPerFile'.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4598462)